t/unit/worker/test_consumer.py

```diff
@@ -306,6 +329,52 @@ def test_cancel_long_running_tasks_on_connection_loss__warning(self):
         with pytest.deprecated_call(match=CANCEL_TASKS_BY_DEFAULT):
             c.on_connection_error_after_connected(Mock())
 
+    def test_ensure_connected_warns_if_deprecated_no_retry_on_startup(self):
+        c = self.get_consumer()
+        c.app.conf.broker_connection_retry_on_startup = None
+        c.app.conf.broker_connection_retry = False
```

Comment:
What should happen if `broker_connection_retry` is True and `broker_connection_retry_on_startup` is still None?

Reply:
In that case, `deprecated_no_retry_on_startup = False`, so the warning is not triggered.

Reply:
That sounds like a bug to me since we want `broker_connection_retry_on_startup` to indicate if we retry connection losses on startup.
What do you think?

Reply:
Maybe we should output a different deprecation warning message instead of not printing anything at all.
We need to let the users know they should set `broker_connection_retry_on_startup` to `True` so they'll have a safe migration path to the next major version.

Reply:
This is a very good idea. I will add the logic and test coverage.

Reply:
Added